vresutils/load.py

```diff
@@ -137,35 +137,47 @@ def timeseries_opsd(years=slice("2011", "2015"), fn=None):
         load = load.loc[years]
 
     # manual alterations:
-    # GB input given in 3 regions, there is no "global one":
+    load = manual_alternations_opsd(load)
+
+    return load
+
+def copy_timeslice(load, cntry, start, stop, delta):
+    start = pd.Timestamp(start)
+    stop = pd.Timestamp(stop)
+    if start in load.index and stop in load.index:
+        load.loc[start:stop, cntry] = load.loc[start+delta:stop+delta, cntry].values
+    return load
+    
+
+def manual_alternations_opsd(load):
+    # GB in the input is split in 3 regions:
     # GBN (Great Britain), NIR (northern ireland), together forming
-    # UKM (united kingdom). Therefore, we choose UKM to be global.
+    # UKM (united kingdom). Therefore, we choose only UKM.
     # the sum of GBN and NIR seems incomplete, more data is missing
     # interpolate the rest (copying from previous weeks might be better)
     load['GB'] = load['GB_UKM']
-    load['GB'] = load['GB'].interpolate()
     load = load.drop(columns=['GB_GBN', 'GB_NIR', 'GB_UKM'])
     
+    # To fill loder periods of gaps (more than 4 hours), we copy a period before into it
+    load = copy_timeslice(load, 'GR', '2015-08-11 21:00', '2015-08-15 20:00', pd.Timedelta(weeks=1))
+    load = copy_timeslice(load, 'AT', '2018-12-31 22:00', '2019-01-01 22:00', pd.Timedelta(days=2))
+    load = copy_timeslice(load, 'CH', '2010-01-19 07:00', '2010-01-19 22:00', pd.Timedelta(days=1))
+    load = copy_timeslice(load, 'CH', '2010-03-28 00:00', '2010-03-28 21:00', pd.Timedelta(days=1))
+    load = copy_timeslice(load, 'CH', '2010-10-08 13:00', '2010-10-10 21:00', pd.Timedelta(days=3))
+    load = copy_timeslice(load, 'CH', '2010-11-04 04:00', '2010-11-04 22:00', pd.Timedelta(days=1))
+    load = copy_timeslice(load, 'NO', '2010-12-09 11:00', '2010-12-09 18:00', pd.Timedelta(days=1))
+    load = copy_timeslice(load, 'GB', '2009-12-31 23:00', '2010-01-31 23:00', pd.Timedelta(weeks=5)) #whole january missing
```

Comment:
Maybe

```suggestion
    load = copy_timeslice(load, 'GB', '2009-12-31 23:00', '2010-01-31 23:00', pd.Timedelta(years=-1)) #whole january missing, so take it from previous year?
```

Reply:
Maybe use the january from the next or previous year, instead of using the february?

Reply:
yes, thats a better idea. will adapt, but years is not a valid keyword, so I made it days=-365. (good idea also to use negative sign, didn't think of that)

Reply:
Then, better take -364 so you get the weekdays correct, as well.